File: topics/arrays/application/read-double.c
double read_double(const char *prompt)
{
    double result = 0;
    
    printf("%s", prompt);
    while (scanf(" %lf", &result) != 1) // Read value, and try to convert double
    {
        // Convert failed, as input was not a number
        scanf("%*[^\n]"); // Read past the end of the current line
        printf("Please enter a number.\n");
        printf("%s", prompt);
    }
    
    return result;
}

File: topics/arrays/application/ReadDouble.pas
function ReadDouble(const prompt: String): Double;
var
    temp: String;
begin
    Write(prompt);
    ReadLn(temp);   // Read the input as a string
    
    while not TryStrToFloat(temp, result) do
    begin
        // Convert failed, as input was not a number
        WriteLn('Please enter a number.');
        
        Write(prompt);
        ReadLn(temp);
    end;
end;